Extensions/xkit_patches.js

```diff
@@ -111,6 +111,29 @@ XKit.tools.get_current_blog = function() {
 	return XKit.tools.get_blogs()[0];
 };
 
+XKit.tools.parse_version = function(versionString) {
+	if (typeof(versionString) === "undefined" || versionString === "") {
+		return {major: 0, minor: 0, patch: 0};
+	}
+	var version = {};
+	var versionSplit = versionString.split(".");
+	if (versionSplit.length < 3) {
+		var versionSplit2 = versionSplit[1].toLowerCase().split("rev");
+		version.major = parseInt(versionSplit[0]);
+		version.minor = parseInt(versionSplit2[0].trim());
+		if (typeof(versionSplit2[1]) === "undefined") {
+			version.patch = 0;
+		} else {
+			version.patch = versionSplit2[1].trim().charCodeAt(0) - "a".charCodeAt(0);
```

Comment:
I think we need to `toLowerCase()` here, because there's no guarantee what case the `rev` will be.